webClient/src/app/shared/editor-control/editor-control.service.ts

```diff
@@ -812,13 +812,18 @@ export class EditorControlService implements ZLUX.IEditor, ZLUX.IEditorMultiBuff
         if (monaco != null) {
           let languages = monaco.languages.getLanguages();
           if (buffer.model.isDataset) {
-            for (let i = 0; i < languages.length; i++) {
-              let lang = languages[i];
-              for (let extension of lang.extensions) {
-                if (name.indexOf(extension)!=-1 || name.indexOf(`${extension}.`) != -1) {
-                  results.push(lang.id);
-                  i = languages.length;
-                  break;
+            //to solve asm.jcllib, search back-to-front, all langs before moving to next portion of name
```

Comment:
Don't have enough datasets experience to know whether this is a good solution. Makes sense to me, but who knows if people would follow the right convention. I guess it's not a problem though since if we get the language wrong, a little future content detection could course-correct us as need be

Reply:
I came up with this, not something I saw from others. One thing I did by convention is lib suffix. Not sure if this is common, but if you put lib at the end, you probably meant to do that.
It's "better" but not as foolproof as reading file contents. Both can be done for higher accuracy.